### PR TITLE
fix replica calculation for cluster capacity expansion

### PR DESCRIPTION
**Pull request by SanjalKatiyar**
https://bugzilla.redhat.com/show_bug.cgi?id=2132733#c17

## Standard deployment:
Screenshot from 2023-05-18 18-15-55

## Non-standard deployment:
Screenshot from 2023-05-18 18-18-42

### CODE REVIEW

**Comment by GowthamShanmugam** [comment]
PR LGTM

File: packages/odf/components/topology/sidebar/storage-cluster/StorageClusterDetails.tsx [hunk 1]
```diff
@@ -138,7 +139,7 @@ export const StorageClusterDetails: React.FC<StorageClusterDetailsProps> = ({
                 isLoading={!nodesLoaded}
                 error={!!nodesLoadError}
                 kind={NodeModel as any}
-                resources={getCephNodes(nodesData as K8sResourceKind[])}
+                resources={getCephNodes(nodesData as NodeKind[])}
```
Comment (bipuladh) [inline packages/odf/components/topology/sidebar/storage-cluster/StorageClusterDetails.tsx:142]:
```suggestion
                resources={getCephNodes(nodesData)}
```
Add type to `watchK8sResource`

**Pull request merged**